python/arcticdb/util/utils.py

```diff
@@ -0,0 +1,325 @@
+        self.__type:TimestampNumber.SupportedFreqTypes = type
+
+
+    def get_type(self) -> SupportedFreqTypes:
+        return self.__type
+
+
+    def get_value(self) -> np.int64:
+        """
+            Returns the value as a number of specified units since Timestamp(0)
+        """
+        return self.value
+
+
+    def to_timestamp(self) -> pd.Timestamp:
+        result, *other = self.calculate_timestamp_after_n_periods(self.value, self.__type)
+        return result
+
+
+    def inc(self, add_number:np.int64) -> 'TimestampNumber':
+        self.value = np.int64(self.value) + np.int64(add_number)
+        return self
+
+
+    def dec(self, add_number:np.int64) -> 'TimestampNumber':
+        self.value = np.int64(self.value) - np.int64(add_number)
+        return self
+
+
+    def to_zero(self) -> 'TimestampNumber':
+        '''
+            To Timestamp(0)
+        '''
+        self.value = 0
+        return self
+    
+
+    def to_initial_value(self) -> 'TimestampNumber':
+        '''
+            Revert to initial value
+        '''
+        self.value = self.init_value
+        return self
+
+
+    def get_initial_value(self) -> 'np.int64':
+        '''
+            Returns the initial value of the number.
+            This allows you to serve like a reference
+        '''
+        return self.init_value
+
+    
+    @classmethod
+    def calculate_timestamp_after_n_periods(cls, periods:int, freq:SupportedFreqTypes='s', 
+            start_time: pd.Timestamp = TIME_ZERO) -> Union[pd.Timestamp, Tuple[pd.Timestamp, pd.Timestamp]]:
+        """ 
+            Calculates end timestamp, based on supplied start timestamp, by adding specified
+            number of time periods denoted by 'freq' parameter ('s' - seconds, 'm' - minutes, 'h' - hours)
+            If periods is negative the end timestamp will be prior to start timestamp
+
+            returns first calculated timestamp and then sorted by time tuple of start time and end time
+        """
+        add=True
+        if (periods < 0):
+            periods:int = -periods
+            add=False
+        
+        if (freq == 's'):
+            if(add):
+                end_time = start_time + pd.Timedelta(seconds=periods) 
+            else:
+                end_time = start_time - pd.Timedelta(seconds=periods) 
+        elif  (freq == 'm'):     
+            if(add):
+                end_time = start_time + pd.Timedelta(minute=periods) 
+            else:
+                end_time = start_time - pd.Timedelta(minute=periods) 
+        elif (freq == 'h'):        
+            if(add):
+                end_time = start_time + pd.Timedelta(hours=periods) 
+            else:
+                end_time = start_time - pd.Timedelta(hours=periods) 
+        else:
+            raise Exception("Not supported frequency")
+
+        if (add):
+            return end_time, (start_time, end_time)
+        else:
+            return end_time, (end_time , start_time)
+
+
+    @classmethod
```

Comment:
The point of staticmethod is to make it object independent but at the same time add it to class that provides utilties.

From there is much easier to access relevant utils and not have to browse through hundreds of methods ot find one you need, example test.py ...

I would think that such kind of grouping utilities by classes would much more speed up work as this forces us to combine similar functions under single class thus would not spread here and there.

At least that is how things are organized in other OOP languages liek C#, Java ... 

Thus perhaps this is a topic for discussion then?

Reply:
We don't normally write code like this. Static methods in Java are a thing because (until very recently) Java forced you to always define code inside of a class. Python and C++ have no such constraint.

I do take the point that it's useful for organisation, so I'm OK with this as-is. But please try to avoid static methods unless there is a good reason